### PR TITLE
Add 3.11 to 3.12 upgrade guide

### PR DESCRIPTION
**Pull request by IKarbowiak**
Add `3.11` to `3.12` upgrade guide, which introduces breaking changes.

### CODE REVIEW

**Comment by peelar** [comment]
to be merged with https://github.com/saleor/saleor-docs/pull/647

File: docs/upgrade-guides/3-11-to-3-12.mdx [hunk 1]
```diff
@@ -0,0 +1,15 @@
+---
+title: Upgrading from 3.11 to 3.12
+---
+
+## Preparation
+
+Version 3.2 introduces some breaking changes necessary to ensure the highest quality of our API.
```
Comment (maarcingebala) [inline docs/upgrade-guides/3-11-to-3-12.mdx:7]:
```suggestion
Version 3.12 introduces some breaking changes necessary to ensure the highest quality of our API.
```

**Review by maarcingebala — APPROVED** [review]

**Review by fowczarek — APPROVED** [review]

**Comment by peelar** [comment]
@IKarbowiak Please provide a link to an example API Reference object field that will be affected by these changes.

**Comment by peelar** [comment]
Moved to https://github.com/saleor/saleor-docs/pull/647.

**Pull request closed without merging**